### PR TITLE
Correct Phone validation for empty values AB#14983

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes or Implements [AB#14983](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14983) (2nd iteration)

## Description
Add async state management in profile and registration form.
Adjusted validation to allow the correct behaviour with empty phone numbers.
Add valid cellphone number for functional test.

## Testing

- [ ] Unit Tests Updated
- [X] Functional Tests Updated
- [ ] Not Required

## UI Changes
Allows empty values to be saved:
image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/views/ProfileView.vue [hunk 1]
```diff
@@ -302,14 +302,15 @@ export default class ProfileView extends Vue {
...
+                return true;
+            }
             if (!ValidationRegEx.PhoneNumberMasked.test(rawInputSmsNumber)) {
                 return false;
             }
             const phoneNumber = PhoneUtil.stripPhoneMask(rawInputSmsNumber);
-            return await this.userProfileService.isPhoneNumberValid(
-                phoneNumber
-            );
+            return this.userProfileService.isPhoneNumberValid(phoneNumber);
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/views/ProfileView.vue:313]: I'm curious why the function was made non-async.

**Review by MikeLyttle — APPROVED** [review]

**Review by MikeLyttle — CHANGES_REQUESTED** [review]
Looks like the unit tests failed.

File: Apps/WebClient/src/ClientApp/src/views/ProfileView.vue (hunk 1 above)
Comment (BronzBierd) [reply Apps/WebClient/src/ClientApp/src/views/ProfileView.vue:313]: When returning the await result Vuelidate's `$pending` wasn't working as expected, when I changed it to return the result from the http request which is also a promise it worked. This works out of the box with Vuelidate so I didn't tweak any further.

https://vuelidate.js.org/#sub-asynchronous-validation

**Comment by BronzBierd** [comment]
Post correction, local unit tests:
image

Moved the null logic to the UserProfileValidator instead of the PhoneNumberValidator which caused the slip.

att: @MikeLyttle

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**